pkg/packages/packages.go

```diff
@@ -0,0 +1,613 @@
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/feeds"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/releases"
+)
+
+var packageOverrideLoopHelpText = heredoc.Doc(`
+bold(PACKAGE SELECTION)
+ This screen presents the list of packages used by your project, and the steps
+ which reference them. 
+ If an item is dimmed (gray text) this indicates that the attribute is duplicated.
+ For example if you reference the same package in two steps, the second will be dimmed. 
+
+bold(COMMANDS)
+ Any any point, you can enter one of the following:
+ - green(?) to access this help screen
+ - green(y) to accept the list of packages and proceed with creating the release
+ - green(u) to undo the last edit you made to package versions
+ - green(r) to reset all package version edits
+ - A package override string.
+
+bold(PACKAGE OVERRIDE STRINGS)
+ Package override strings must have 2 or 3 components, separated by a :
+ The last component must always be a version number.
+ 
+ When specifying 2 components, the first component is either a Package ID or a Step Name.
+ You can also specify a * which will match all packages
+ Examples:
+   bold(octopustools:9.1)   dim(# sets package 'octopustools' in all steps to v 9.1)
+   bold(Push Package:3.0)   dim(# sets all packages in the 'Push Package' step to v 3.0)
+   bold(*:5.1)              dim(# sets all packages in all steps to v 5.1)
+
+ The 3-component syntax is for advanced use cases where you reference the same package twice
+ in a single step, and need to distinguish between the two.
+ The syntax is bold(packageIDorStepName:packageReferenceName:version)
+ Please refer to the octopus server documentation for more information regarding package reference names. 
+
+dim(---------------------------------------------------------------------)
+`) // note this expects to have prettifyHelp run over it
+
+type StepPackageVersion struct {
+	// these 3 fields are the main ones for showing the user
+	PackageID  string
+	ActionName string // "StepName is an obsolete alias for ActionName, they always contain the same value"
+	Version    string // note this may be an empty string, indicating that no version could be found for this package yet
+
+	// used to locate the deployment process VersioningStrategy Donor Package
+	PackageReferenceName string
+}
+
+// BuildPackageVersionBaseline loads the deployment process template from the server, and for each step+package therein,
+// finds the latest available version satisfying the channel version rules. Result is the list of step+package+versions
+// to use as a baseline. The package version override process takes this as an input and layers on top of it
+func BuildPackageVersionBaseline(octopus *octopusApiClient.Client, packages []releases.ReleaseTemplatePackage,
+	setAdditionalFeedQueryParameters func(releases.ReleaseTemplatePackage, feeds.SearchPackageVersionsQuery) (feeds.SearchPackageVersionsQuery, error)) ([]*StepPackageVersion, error) {
```

Comment:
This `func` parameter has a change in it to make it generic and usable when creating Git runbooks and creating releases. When creating a release we need to check the channel rules for the associated package and add any additional search parameters to ensure that the version selected satisfies the rules. This parameter allows us to do that.